### PR TITLE
pause and resume bugfix.

### PR DESCRIPTION
**Pull request by kseniaxu0704**
The Pause and resume buttons can not be set the tomorrow as the reactivation date after 16:00 o'clock.
image
For example:

Today is Tue 24 January 2023, 21:48:12 in PTS 
 if I select 1/25 as the reactivation date, if use the old code;
The reactivates day is  UTC timezone (Wed Jan 25, 2023, 00:00:00), local Time Zone(Tue 24 January 2023 16:00:00, Milliseconds is 1674604800000, 
And the current time is UTC timezone (Wed Jan 25, 2023, 05:48:12), local Time Zone(Tue 24 January 2023 21:48:12), Milliseconds is 1674625692694
And the Date. parse(activation date) < Date. now(), therefore it is impossible to select the tomorrow as reactivate date.
Reason to cause that: In a world, it is different timezone problem.
Method:
False:
The reactivates day is  UTC timezone (Wed Jan 25, 2023, 00:00:00), local Time Zone(Tue 24 January 202316:00:00, Milliseconds is 1674604800000
True:
The reactivates day is  UTC timezone (Wed Jan 25 2023 08:00:00), local Time Zone(Wed Jan 25, 2023, 00:00:00), Milliseconds is 1674633600000

Above effect can achieve by package moment()
moment(date), moment() is local time compare to local time.

### CODE REVIEW

**Comment by rauleffting** [comment]
Please, @kseniaxu0704, explain how to test it properly.  Thank you!

**Comment by RafaelCastilhoCastro** [comment]
Hi @kseniaxu0704,
Sorry I couldn't understand properly your suggested solution. Does Moment package solve the timezone problem completely? Please provide some "How to test" steps so we can help. Like it is I don't know how to test it.
Thanks.

File: src/components/UserManagement/ActivationDatePopup.jsx [hunk 1]
```diff
@@ -13,7 +14,7 @@ const ActivationDatePopup = React.memo(props => {
     props.onClose();
   };
   const pauseUser = () => {
-    if (Date.parse(activationDate) > Date.now()) {
+    if (moment().isBefore(moment(activationDate))) {
```
Comment (LeeSingChanYau) [inline src/components/UserManagement/ActivationDatePopup.jsx:17]: So activationDate also information of what timezone the user was when he paused it? Please provide steps to test this PR.

**Review by LeeSingChanYau — COMMENTED** [review]
So activationDate also information of what timezone the user was when he paused it? Please provide steps to test this PR.

**Comment by kseniaxu0704** [comment]
> Please, @kseniaxu0704, explain how to test it properly. Also, remember to post it on Slack. Thank you!

Please pull this pr, go to dashboard - view profile - pause/resum button
image
select tomorrow as reactivation day 
to verify it work or not. 
Thanks

**Comment by kseniaxu0704** [comment]
> 

Please pull this pr, go to dashboard - view profile - pause/resum button
image
select tomorrow as reactivation day
to verify its works or not.
Thanks

File: src/components/UserManagement/ActivationDatePopup.jsx (hunk 1 above)
Comment (kseniaxu0704) [reply src/components/UserManagement/ActivationDatePopup.jsx:17]: image
When user select the date , (moment().isBefore(moment(activationDate)) will compare the user's local time and the date in local time zone.

**Comment by kseniaxu0704** [comment]
> 

Please pull this pr, go to dashboard - view profile - pause/resum button
image
select tomorrow as reactivation day
to verify its works or not.
Thanks

**Review by rauleffting — DISMISSED** [review]
Hi @kseniaxu0704 . It worked perfectly, as you can see below:

https://user-images.githubusercontent.com/29555732/215526639-d6f74af6-ebb0-4c61-8c71-61bc454914c0.mov

Also, the code is very simple. You just used the moment library. I have nothing to ask you about it. Thank you very much!

**Comment by RafaelCastilhoCastro** [comment]
Hi @kseniaxu0704,

I tested the feature and it seems to been working as intended. Thanks.

**Comment by AnishPandita** [comment]
Hey @kseniaxu0704 , I tested your changes and would everything working fine as you can see below. The overall code looks nice and readable. Usage of moment package to handle the timezone issue was a great solution.

https://user-images.githubusercontent.com/111321945/215668159-544537b6-f026-4751-a562-bc6d0bc80ab1.mov

**Comment by Gukal7** [comment]
Hey @kseniaxu0704, The bugfix is working just fine. I had a look at the code as well and it looks clean.
But I had a question/doubt, shouldn't the prompt say Paused until *the date you select*. For example in my case it should say Paused until 2023-02-01 but it's showing Paused until 2023-01-31. I've selected 1 Feb as the reactivation date. I am suggesting because the word "until" is used which means "up to the point in time or the event mentioned."

Screenshot 2023-01-31 at 3 33 17 PM
Screenshot 2023-01-31 at 3 33 39 PM

**Comment by one-community** [comment]
Good catch Gurkanwal! Yan, please update the world "until" to say "through" instead: Paused through 2023-01-31. 

Please also change the format of those dates to the standard for the US: MM-DD-YYYY

**Comment by Filipe-Oliv** [comment]
Hi @kseniaxu0704 

Everything works fine. The only thing is the same as @one-community and @Gukal7 talked to you about changing 'until' to 'through'.

[week 03 - 02-07-23 (7).webm](https://user-images.githubusercontent.com/114119668/217376479-6c21a2f9-f05b-4162-9ee5-4db2251ed42b.webm)

**Comment by kseniaxu0704** [comment]
> Hey @kseniaxu0704, The bugfix is working just fine. I had a look at the code as well and it looks clean. But I had a question/doubt, shouldn't the prompt say Paused until _the date you select_. For example in my case it should say Paused until 2023-02-01 but it's showing Paused until 2023-01-31. I've selected 1 Feb as the reactivation date. I am suggesting because the word "until" is used which means "up to the point in time or the event mentioned."
> 
> Screenshot 2023-01-31 at 3 33 17 PM
> 
> Screenshot 2023-01-31 at 3 33 39 PM

image
Has been update as your suggestion

**Comment by kseniaxu0704** [comment]
> Good catch Gurkanwal! Yan, please update the world "until" to say "through" instead: Paused through 2023-01-31.
> 
> Please also change the format of those dates to the standard for the US: MM-DD-YYYY

image
Has been update as suggestion

**Comment by kseniaxu0704** [comment]
> Hi @kseniaxu0704
> 
> Everything works fine. The only thing is the same as @one-community and @Gukal7 talked to you about changing 'until' to 'through'.
> 
>  week.03.-.02-07-23.7.webm

image
has been update as comment

**Comment by kseniaxu0704** [comment]
> Hi @kseniaxu0704
> 
> Everything works fine. The only thing is the same as @one-community and @Gukal7 talked to you about changing 'until' to 'through'.
> 
>  week.03.-.02-07-23.7.webm

image
has been update as comment, please test

**Comment by RafaelCastilhoCastro** [comment]
Hi @kseniaxu0704,
I tested and the corrections are working as intended. The code is neat. Thanks.

**Review by RafaelCastilhoCastro — APPROVED** [review]

**Pull request merged**